libs/data-types/src/lib/utilities/getReferencesForCoscradDataSchema.ts

```diff
@@ -0,0 +1,30 @@
+import { isNullOrUndefined } from '@coscrad/validation-constraints';
+import { ClassSchema } from '../types';
+
+export interface ReferenceSpecification {
+    type: string;
+    path: string;
+    isArray: boolean;
+}
+
+/**
+ * TODO We need to recurse and support nested references.
+ *
+ * Rename foo!
+ */
+export const getReferencesForCoscradDataSchema = (foo: ClassSchema<Record<string, unknown>>) =>
+    Object.entries(foo).reduce((acc: ReferenceSpecification[], [propertyKey, typeDefinition]) => {
+        // @ts-expect-error TODO fix types
```

Comment:
Can we fix the types here?

Reply:
This is a tough one. We need to tighten up type safety of the `@coscrad/data-types` lib.